x/mongo/driver/options/options.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) MongoDB, Inc. 2025-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package options
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/v2/mongo/options"
+	"go.mongodb.org/mongo-driver/v2/x/mongo/driver"
+)
+
+// SetInternalClientOptions sets internal options for ClientOptions.
+//
+// Deprecated: This function is for internal use only. It may be changed or removed in any release.
+func SetInternalClientOptions(opts *options.ClientOptions, key string, option any) error {
```

Comment:
Suggest updating this now to support the custom field and extending the compilecheck to show that it doesn't break anything:

```go
package main

import (
	"fmt"

	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
	"go.mongodb.org/mongo-driver/v2/mongo/options"
	"go.mongodb.org/mongo-driver/v2/x/mongo/driver/options"
)

func main() {
	opts := options.Client()
	options.SetInternalClientOptions(opts, "foo", "bar")

	_, _ = mongo.Connect(options.Client())
	fmt.Println(bson.D{{Key: "key", Value: "value"}})
}
```